mpt/impl/imap-mailbox/external-james/src/test/java/org/apache/james/mpt/imapmailbox/external/james/DockerDeploymentValidationTest.java

```diff
@@ -0,0 +1,74 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mpt.imapmailbox.external.james;
+
+import org.apache.james.mpt.api.ImapHostSystem;
+import org.apache.james.mpt.imapmailbox.external.james.host.DomainsAndUsersAdder;
+import org.apache.james.mpt.imapmailbox.external.james.host.SmtpHostSystem;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.testcontainers.containers.Container;
+
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+
+public class DockerDeploymentValidationTest extends DeploymentValidation {
+
+    private ImapHostSystem system;
+    private SmtpHostSystem smtpHostSystem;
+
+    @Rule
+    public DockerJamesRule dockerJamesRule = new DockerJamesRule("linagora/james-jpa-guice");
+
+    @Override
+    @Before
+    public void setUp() throws Exception {
+
+        dockerJamesRule.start();
+
+        DomainsAndUsersAdder domainsAndUsersAdder = dockerJamesRule.cliJarDomainsAndUsersAdder();
+        Injector injector = Guice.createInjector(new DeploymentValidationJamesModule(dockerJamesRule.getConfiguration(), domainsAndUsersAdder));
+        domainsAndUsersAdder.addDomain(DOMAIN);
+        domainsAndUsersAdder.addUser(USER_ADDRESS, PASSWORD);
+        system = injector.getInstance(ImapHostSystem.class);
+        smtpHostSystem = injector.getInstance(SmtpHostSystem.class);
```

Comment:
what is the reason to create a guice injector to only initialize `SmtpHostSystem` and `ImapHostSystem`? We can very well create them directly, right?

Reply:
ImapHostSystem was already injected that way.
So we have done the same for the SmtpHostSystem. For the sake of consistency.